src/runtime/pmix_params.c

```diff
@@ -157,6 +157,13 @@ pmix_status_t pmix_register_params(void)
                                        PMIX_INFO_LVL_1, PMIX_MCA_BASE_VAR_SCOPE_ALL,
                                        &pmix_client_globals.base_verbose);
 
+    pmix_client_globals.get_fastpath = true;
+    (void) pmix_mca_base_var_register ("pmix", "pmix", "client", "get_fastpath",
+                                       "Enable fastpath for client get operations",
+                                       PMIX_MCA_BASE_VAR_TYPE_BOOL, NULL, 0, 0,
+                                       PMIX_INFO_LVL_1, PMIX_MCA_BASE_VAR_SCOPE_ALL,
+                                       &pmix_client_globals.get_fastpath);
```

Comment:
I assume this means that the user is declaring their app to be single-threaded? We looked at that case with an eye to removing threadshifting on inbound API calls, but ran into a problem as it still meant that the user's thread might be simultaneously accessing an object inside PMIx while the PMIx progress thread was also accessing it. See #951, #956, and #959 for one thread on the issue.

Seems to me you have some similar issues here. Even if you proposed code path stays inside the user's thread, you still have the problem of the PMIx progress thread touching overlapping code paths (e.g., in the dstore) stemming from requests for GDS data from other parts of the of the user's code. For example, a call to ```PMIx_Query_info_nb```.

I'm honestly not sure I see how we could make this code path work without turning "off" the PMIx internal progress thread - which would almost certainly break quite a few things.

Reply:
No, users are not required to be single-threaded. Dstore protects it's state.

Reply:
PMIx progress thread will also be protected with locks.